pyogrio/tests/test_geopandas_io.py

```diff
@@ -186,6 +186,79 @@ def test_read_fids_force_2d(test_fgdb_vsi):
         assert not df.iloc[0].geometry.has_z
 
 
+#@pytest.mark.filterwarnings("ignore: Layer")
+def test_read_sql(naturalearth_lowres):
+    # empty filter should return full set of records
+    df = read_dataframe(naturalearth_lowres, sql=None)
+    assert len(df) == 177
+    assert len(df.columns) == 6
+
+    # should return singular item
+    sql = "SELECT * FROM naturalearth_lowres WHERE iso_a3 = 'CAN'"
+    df = read_dataframe(naturalearth_lowres, sql=sql)
+    assert len(df) == 1
+    assert len(df.columns) == 6
+    assert df.iloc[0].iso_a3 == "CAN"
+    area_canada = df.iloc[0].geometry.area
+
+    # use spatialite function (needs SQLITE dialect)
```

Comment:
Suggest putting `sqlite` dialect test into a separate test case `test_read_sql_dialect`

Reply:
I put the sqlite test in a seperate function, but I already had a go at testing on different file formats and for that I had to specify the OGRSQL dialect on the other tests. If the dialect is not specified (dialect=None), it will depend on the database/file you connect to which dialect will be used:
- shapefile -> OGRSQL
- gpkg -> SQLite
(- postgress -> postgress)
(- Oracle -> oracle)
(- ...)

So I named the test function test_read_sql_dialect_sqlite.

Reply:
Ok - that will be a good thing to note in the documentation, since I would have naively expected it to default to `OGRSQL` unless directed otherwise.